plugins/atomicfu/atomicfu-compiler/test/org/jetbrains/kotlinx/atomicfu/JsGenerateAndRunTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.atomicfu
+
+import org.jetbrains.kotlin.cli.common.config.KotlinSourceRoot
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.jetbrains.kotlin.js.config.JSConfigurationKeys
+import org.jetbrains.kotlin.js.test.BasicBoxTest
+import org.jetbrains.kotlin.js.test.BasicIrBoxTest
+import org.jetbrains.kotlin.serialization.js.ModuleKind
+import org.jetbrains.kotlin.test.KotlinTestUtils
+import org.jetbrains.kotlinx.atomicfu.compiler.extensions.AtomicfuComponentRegistrar
+import org.junit.Test
+import java.io.File
+
+private val atomicfuCompileDependency = System.getProperty("atomicfu.classpath")
+private val atomicfuRuntime = System.getProperty("atomicfuRuntimeForTests.classpath")
+
+abstract class AtomicfuBaseTest(relativePath: String) : BasicIrBoxTest(
```

Comment:
Could we implement an isolated runner for such tests? I could be much less complicated than Box test runner